### PR TITLE
feat(variant_mapper, bigcommerce_product_agent): list of not purchasable

### PR DESCRIPTION
**Pull request by Jhalter5Stones**
list of not purchasable fields get set to disable purchasable field

### CODE REVIEW

**Comment by Jhalter5Stones** [comment]
@mrogers-5s No rush here but whenever you get the chance can you review. Thank you!

File: lib/huginn_bigcommerce_product_agent/bigcommerce_product_agent.rb [hunk 1]
```diff
@@ -62,6 +63,10 @@ def validate_options
             unless options['mode'].present? && modes.include?(options['mode'])
                 errors.add(:base, "mode is a required field and must be one of: #{modes.join(', ')}")
             end
+
+            unless options['not_purchasable_format_list'].present?
+                errors.add(:base, 'not_purchasable_format_list is a required field')
+            end
```
Comment (mrogers-5s) [inline lib/huginn_bigcommerce_product_agent/bigcommerce_product_agent.rb:69]: @Jhalter5Stones If the default options use an empty array, then this shouldn't be required.

If you have logic that expects an array later in the code, default it to an empty array there if it's not provided in the config

**Comment by Jhalter5Stones** [comment]
@mrogers-5s Ok I think I fixed the issue of not validating that not_purchasable_format_list is an array. Let me know what you think

File: lib/huginn_bigcommerce_product_agent/bigcommerce_product_agent.rb [hunk 2]
```diff
@@ -62,6 +63,10 @@ def validate_options
             unless options['mode'].present? && modes.include?(options['mode'])
                 errors.add(:base, "mode is a required field and must be one of: #{modes.join(', ')}")
             end
+
+            unless options['not_purchasable_format_list'].is_a?(Array)
+                errors.add(:base, 'not_purchasable_format_list is a required field')
+            end
```
Comment (mrogers-5s) [inline lib/huginn_bigcommerce_product_agent/bigcommerce_product_agent.rb:69]: @Jhalter5Stones I would make sure the error message actually reflects the validation.

Additionally, I would make sure that you only check whether the option is an array if the option is provided

Comment (mrogers-5s) [reply lib/huginn_bigcommerce_product_agent/bigcommerce_product_agent.rb:69]:
```
if options[...].present? && !options[...].is_a?(Array)
    errors.add( ... )
end
```

**Comment by Jhalter5Stones** [comment]
@Ken-Michalak Ready for final review

https://trello.com/c/yr3N944q/182-usbat-only-see-purchase-options-for-video-and-a-button-to-watch-on-formed-when-available

**Comment by Ken-Michalak** [comment]
@mrogers-5s still needs to approve it.

**Comment by Jhalter5Stones** [comment]
@Ken-Michalak Sorry. He approved it over chat and I forgot that he has to actually approve it in github for it to go through

@mrogers-5s Can you click approve?

**Pull request merged**